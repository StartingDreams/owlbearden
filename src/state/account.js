import { fromJS } from 'immutable';

export const firebaseConfig = {
  apiKey: 'AIzaSyDg4Z-8ROBMuRtzU5ThnU1GMd2IWgsttEU',
  authDomain: 'owlbearsden.firebaseapp.com',
  databaseURL: 'https://owlbearsden.firebaseio.com',
  projectId: 'owlbearsden',
  storageBucket: 'owlbearsden.appspot.com',
  messagingSenderId: '149275717879',
};

export const Types = {
  LOGIN: 'owlbearden/LOGIN',
  LOGOUT: 'owlbearden/LOGOUT',
};

export const login = user => ({
  type: Types.LOGIN,
  user,
});

export const logout = () => ({
  type: Types.LOGOUT,
});

export const INITIAL_STATE = fromJS({
  user: null,
  isAuthenticated: false,
});

export default function reducer(state = INITIAL_STATE, action = {}) {
  switch (action.type) {
    case Types.LOGIN:
      return state.withMutations(s => s
        .set('user', fromJS(action.user))
        .set('isAuthenticated', true));
    case Types.LOGOUT:
      return state.withMutations(s => s
        .set('user', fromJS(action.user))
        .set('isAuthenticated', false));
    default:
      return state;
  }
}
