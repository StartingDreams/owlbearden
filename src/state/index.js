import { combineReducers } from 'redux';
import configureStore from './CreateStore';
import ui from './ui';
import casting from '../modules/Casting/ducks/CastingRedux';

export default () => {
  const rootReducer = combineReducers({
    ui,
    casting,
  });

  return configureStore(rootReducer);
};
