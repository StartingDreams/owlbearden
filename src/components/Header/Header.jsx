import React from 'react';
import AppBar from 'material-ui/AppBar';
import FlatButton from 'material-ui/FlatButton';
import IconButton from 'material-ui/IconButton';
import FontIcon from 'material-ui/FontIcon';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import firebase from 'firebase';
import { withRouter } from 'react-router-dom';
import { toggleLeftDrawer, closeLeftDrawer } from '../../state/ui';
import { logout } from '../../state/auth';

export class Header extends React.Component {
  toggleLeftDrawerClick = () => {
    const { dispatch } = this.props;
    dispatch(toggleLeftDrawer());
  };

  closeLeftDrawerClick = () => {
    const { dispatch } = this.props;
    dispatch(closeLeftDrawer());
  };

  logoutClick = async () => {
    await firebase.auth().signOut();
    const { dispatch } = this.props;
    dispatch(logout());
    this.props.history.push('/login');
  };

  navigateHome = () => {
    this.closeLeftDrawerClick();
    this.props.history.push('/dashboard');
  };

  render = () => {
    const { isAuthenticated } = this.props;
    return (
      <AppBar
        title={<span style={{ cursor: 'pointer' }}>{'Owlbear\'s Den'}</span>}
        iconElementLeft={<IconButton><FontIcon className="material-icons">menu</FontIcon></IconButton>}
        onLeftIconButtonTouchTap={this.toggleLeftDrawerClick}
        onTitleTouchTap={this.navigateHome}
        iconElementRight={isAuthenticated ? (<FlatButton id="head-logout-button" label="Logout" onClick={this.logoutClick} />) : (<span />)}
      />
    );
  };
}

Header.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  dispatch: PropTypes.func,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export const mapStateToProps = state => ({
  isAuthenticated: state.auth.get('isAuthenticated'),
});

export default withRouter(connect(
  mapStateToProps,
)(Header));
