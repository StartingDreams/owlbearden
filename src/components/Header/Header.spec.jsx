import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { fromJS } from 'immutable';
import { Header, mapStateToProps } from './Header';
import { toggleLeftDrawer, closeLeftDrawer } from '../../state/ui';
import { logout } from '../../state/auth';

jest.mock('firebase', () => {
  const signOut = jest.fn();
  const authReturn = {
    signOut,
  };
  const auth = () => (authReturn);

  auth.signOut = signOut;
  return { auth };
});
const firebase = require('firebase');

describe('<Header />', () => {
  let isAuthenticated = null;
  let dispatch = null;
  let history = null;
  beforeEach(() => {
    isAuthenticated = true;
    dispatch = jest.fn();
    history = {
      push: jest.fn(),
    };
  });

  it('should match the authenticated snapshot', () => {
    const wrapper = shallow(
      <Header
        isAuthenticated
        dispatch={dispatch}
        history={history}
      />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should match the unauthenticated snapshot', () => {
    const wrapper = shallow(
      <Header
        isAuthenticated={false}
        dispatch={dispatch}
        history={history}
      />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('logout should be dispatched', async () => {
    const wrapper = shallow(
      <Header
        isAuthenticated={isAuthenticated}
        dispatch={dispatch}
        history={history}
      />);
    await wrapper.instance().logoutClick();
    expect(history.push).toHaveBeenCalledWith('/login');
    expect(dispatch).toHaveBeenCalledWith(logout());
  });
  it('closeLeftDrawer should be dispatched', () => {
    const wrapper = shallow(
      <Header
        isAuthenticated={isAuthenticated}
        dispatch={dispatch}
        history={history}
      />);
    wrapper.instance().closeLeftDrawerClick();
    expect(dispatch).toHaveBeenCalledWith(closeLeftDrawer());
  });
  it('toggleLeftDrawer should be dispatched', () => {
    const wrapper = shallow(
      <Header
        isAuthenticated={isAuthenticated}
        dispatch={dispatch}
        history={history}
      />);
    wrapper.instance().toggleLeftDrawerClick();
    expect(dispatch).toHaveBeenCalledWith(toggleLeftDrawer());
  });
  it('navigateHome should go to dashboard', () => {
    const wrapper = shallow(
      <Header
        isAuthenticated={isAuthenticated}
        dispatch={dispatch}
        history={history}
      />);
    wrapper.instance().navigateHome();
    expect(history.push).toHaveBeenCalledWith('/dashboard');
  });
  it('mapStateToProps - isAuthenticated should be passed in', () => {
    const isAuthenticated = false;
    const auth = fromJS({
      isAuthenticated,
    });
    const state = {
      auth,
    };
    const props = mapStateToProps(state);
    expect(props.isAuthenticated).toBe(auth.get('isAuthenticated'));
  });
});
