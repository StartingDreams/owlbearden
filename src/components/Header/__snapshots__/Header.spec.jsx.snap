// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match the authenticated snapshot 1`] = `
<AppBar
  iconElementLeft={
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      tooltipPosition="bottom-center"
      touch={false}
    >
      <FontIcon
        className="material-icons"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        menu
      </FontIcon>
    </IconButton>
  }
  iconElementRight={
    <FlatButton
      disabled={false}
      fullWidth={false}
      id="head-logout-button"
      label="Logout"
      labelPosition="after"
      labelStyle={Object {}}
      onClick={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
    />
  }
  onLeftIconButtonTouchTap={[Function]}
  onTitleTouchTap={[Function]}
  showMenuIconButton={true}
  title={
    <span
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Owlbear's Den
    </span>
  }
  zDepth={1}
/>
`;

exports[`<Header /> should match the unauthenticated snapshot 1`] = `
<AppBar
  iconElementLeft={
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      tooltipPosition="bottom-center"
      touch={false}
    >
      <FontIcon
        className="material-icons"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        menu
      </FontIcon>
    </IconButton>
  }
  iconElementRight={<span />}
  onLeftIconButtonTouchTap={[Function]}
  onTitleTouchTap={[Function]}
  showMenuIconButton={true}
  title={
    <span
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Owlbear's Den
    </span>
  }
  zDepth={1}
/>
`;
